app/scripts/contentscript.js

```diff
@@ -391,23 +492,27 @@ function logStreamDisconnectWarning(remoteLabel, error) {
 
 /**
  * The function send message to inpage to notify it of extension stream connection
+ * This is used as notification to replay any pending messages in MV3
  */
-function notifyInpageOfExtensionStreamConnect() {
-  window.postMessage(
-    {
-      target: INPAGE, // the post-message-stream "target"
-      data: {
-        // this object gets passed to obj-multiplex
-        name: PROVIDER, // the obj-multiplex channel name
-        data: {
-          jsonrpc: '2.0',
-          method: 'METAMASK_EXTENSION_STREAM_CONNECT',
-        },
-      },
-    },
-    window.location.origin,
-  );
-}
+// function extensionStreamMessageListener(msg) {
```

Comment:
the commented out lines related to `extensionStreamMessageListener` will be handled by @jpuri's PR here: https://github.com/MetaMask/metamask-extension/pull/16250

Reply:
Perhaps adding a `//TODO: ` with a reference is warranted?  :)